test/ret_web/controllers/api-internal/change_login_email_controller_test.exs

```diff
@@ -0,0 +1,74 @@
+defmodule RetWeb.ApiInternal.V1.ChangeLoginEmailControllerTest do
+  use RetWeb.ConnCase
+  import Ret.TestHelpers
+  alias Ret.Account
+
+  @dashboard_access_header "x-ret-dashboard-access-key"
+  @dashboard_access_key "test-key"
+
+  setup_all do
+    merge_module_config(:ret, RetWeb.Plugs.DashboardHeaderAuthorization, %{dashboard_access_key: @dashboard_access_key})
+
+    on_exit(fn ->
+      Ret.TestHelpers.merge_module_config(:ret, RetWeb.Plugs.DashboardHeaderAuthorization, %{dashboard_access_key: nil})
+    end)
+  end
+
+  test "new email addresses must be valid", %{conn: conn} do
```

Comment:
Consider describing the function under test. In controller tests I like to wrap tests for a given action in a describe like so:

```elixir
describe "PUT update" do
  ...
end
```